beacon_node/beacon_chain/src/custody_context.rs

```diff
@@ -150,6 +150,24 @@ impl ValidatorRegistrations {
                 .or_insert(latest_validator_custody);
         }
     }
+
+    /// Updates the `epoch -> cgc` map by pruning records before `effective_epoch`
+    /// while setting the `cgc` at `effective_epoch` to the latest validator custody requirement.
+    ///
+    /// This is used to restart custody backfill sync at `effective_epoch`
+    pub fn reset_validator_custody_requirements(&mut self, effective_epoch: Epoch) {
+        if let Some(latest_validator_custody_requirements) =
+            self.latest_validator_custody_requirement()
+        {
+            self.epoch_validator_custody_requirements
+                .retain(|&epoch, _| epoch >= effective_epoch);
+
+            self.epoch_validator_custody_requirements
+                .entry(effective_epoch)
+                .and_modify(|old_custody| *old_custody = latest_validator_custody_requirements)
+                .or_insert(latest_validator_custody_requirements);
```

Comment:
Isn't this just `.insert`?

```suggestion
            self.epoch_validator_custody_requirements
                .insert(effective_epoch, latest_validator_custody_requirements)
```

Reply:
ah yeah you're right, i'll also update the identical code in the `backfill_validator_custody_requirements` function as  well

Reply:
Addressed in https://github.com/sigp/lighthouse/pull/8339/commits/580b584a8f0838906171a421c3b404ae8a905e07